pkg/controller/machinehealthcheck/machinehealthcheck_controller.go

```diff
@@ -331,19 +331,13 @@ func hasMatchingLabels(machineHealthCheck *healthcheckingv1alpha1.MachineHealthC
 }
 
 func isMaster(machine mapiv1.Machine, client client.Client) bool {
-	machineMasterLabels := []string{
-		"machine.openshift.io/cluster-api-machine-role",
-		"machine.openshift.io/cluster-api-machine-type",
-		"sigs.k8s.io/cluster-api-machine-role",
-		"sigs.k8s.io/cluster-api-machine-type",
-	}
-	nodeMasterLabels := []string{
+	masterLabels := []string{
```

Comment:
Does it make sense to check machine labels at all?
IMHO the only proper way is to check node labels that machine reference to.

Reply:
I agree. @vikaschoudhary16 as part of this effort can we drop labels here https://github.com/openshift/installer/tree/master/pkg/asset/machines/aws
And update https://github.com/openshift/origin/pull/22090/files#r267287918 to fetch from node accordingly cc @ingvagabund @derekwaynecarr